test/common/http/hcm_router_fuzz_test.cc

```diff
@@ -0,0 +1,575 @@
+    ON_CALL(mock_route_->route_entry_, clusterName()).WillByDefault(ReturnRef(name_));
+    ON_CALL(mock_route_->route_entry_.internal_redirect_policy_, enabled())
+        .WillByDefault(Return(internal_redirect_policy_enabled_));
+    ON_CALL(mock_route_->route_entry_.internal_redirect_policy_, shouldRedirectForResponseCode(_))
+        .WillByDefault(
+            Invoke([](const Http::Code& code) { return code == Http::Code::MovedPermanently; }));
+    ON_CALL(mock_route_->route_entry_.internal_redirect_policy_, maxInternalRedirects())
+        .WillByDefault(Return(10));
+    ON_CALL(mock_route_->route_entry_.internal_redirect_policy_, isCrossSchemeRedirectAllowed())
+        .WillByDefault(Return(cross_scheme_redirect_allowed_));
+    ON_CALL(mock_route_->route_entry_.early_data_policy_, allowsEarlyDataForRequest(_))
+        .WillByDefault(Return(allows_early_data_for_request_));
+    ON_CALL(*tlc_.cluster_.info_.get(), maintenanceMode()).WillByDefault(Return(maintenance_));
+    tlc_.cluster_.info_->upstream_config_ =
+        std::make_unique<envoy::config::core::v3::TypedExtensionConfig>();
+    tlc_.cluster_.info_->upstream_config_->set_name("envoy.filters.connection_pools.http.generic");
+  }
+
+  void newUpstream(Router::GenericConnectionPoolCallbacks* request,
+                   absl::optional<Envoy::Http::Protocol> protocol) {
+    auto upstream = std::make_unique<FuzzUpstream>(request->upstreamToDownstream());
+    auto stream = std::make_unique<Extensions::Upstreams::Http::Http::HttpUpstream>(
+        request->upstreamToDownstream(), &upstream->mock_request_encoder_);
+    Upstream::HostDescriptionConstSharedPtr host =
+        std::make_shared<Upstream::MockHostDescription>();
+    request->onPoolReady(std::move(stream), host, mock_stream_info_.downstreamAddressProvider(),
+                         mock_stream_info_, protocol);
+    upstreams_.push_back(std::move(upstream));
+  }
+
+  void sendHeaders(uint32_t stream, const test::fuzz::Headers& response_headers, bool end_stream) {
+    FuzzUpstream* s = select(stream);
+    if (s) {
+      auto headers = std::make_unique<Http::TestResponseHeaderMapImpl>(
+          Fuzz::fromHeaders<Http::TestResponseHeaderMapImpl>(response_headers, {}, {"status"}));
+      uint64_t rc;
+      if (!absl::SimpleAtoi(headers->getStatusValue(), &rc)) {
+        headers->setStatus(302);
+      }
+      s->sendHeaders(std::move(headers), end_stream);
+    }
+  }
+
+  void sendData(uint32_t stream, const std::string& data, bool end_stream) {
+    FuzzUpstream* s = select(stream);
+    if (s) {
+      Buffer::OwnedImpl buf(data);
+      s->sendData(buf, end_stream);
+    }
+  }
+
+  void sendTrailers(uint32_t stream, const test::fuzz::Headers& response_trailers) {
+    FuzzUpstream* s = select(stream);
+    if (s) {
+      auto trailers = std::make_unique<Http::TestResponseTrailerMapImpl>(
+          Fuzz::fromHeaders<Http::TestResponseTrailerMapImpl>(response_trailers, {}, {"status"}));
```

Comment:
Is `status` needed as part of the trailers?

Reply:
no, removed.